client/my-sites/email/email-providers-comparison/stacked/index.tsx

```diff
@@ -86,12 +92,47 @@ const EmailProvidersStackedComparison = ( {
 		)
 	);
 
+	const currentUserCanAddEmail = canCurrentUserAddEmail( domain );
+
+	const ownerUserName = useDomainOwnerUserName( selectedSite, domain );
+
+	const isPrivacyAvailable = domain?.privacyAvailable;
+
+	const contactOwnerUrl = `https://privatewho.is/?s=${ selectedDomainName }`;
+
+	const loginUrl = loginUrlWithUserNameAndRedirectToEmailProvidersComparison(
+		ownerUserName,
+		selectedSite?.slug,
+		selectedDomainName
+	);
+
+	/**
+	 * This function builds the event 'calypso_email_providers_owner_contact_click'
+	 * or 'calypso_email_providers_user_login_click' depending on the event type passed as parameter.
+	 *
+	 * @param { "owner_contact" | "user_login" } eventType
+	 */
+	const onClickLink = ( eventType: 'owner_contact' | 'user_login' ) => {
+		dispatch(
+			recordTracksEvent( `calypso_email_providers_${ eventType }_click`, {
+				owner_login: ownerUserName,
+			} )
+		);
```

Comment:
Even if it means more code, it's usually better to avoid building event names dynamically as that makes it impossible to grok them later:

```suggestion
		let eventName = `calypso_email_providers_owner_contact_click`;
		
		if ( some logic ) {
			eventName = `calypso_email_providers_user_login_click`;
		}
		
		dispatch(
			recordTracksEvent( eventName , {
				owner_login: ownerUserName,
			} )
		);
```

Reply:
That was the intention of the superfluous documentation :) But I will do as you suggest :)